src/Integration/NewConnectedMode/IConfigurationProvider.cs

```diff
@@ -18,12 +18,9 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using SonarLint.VisualStudio.Core.Binding;
-
 namespace SonarLint.VisualStudio.Integration.NewConnectedMode
 {
-    public interface IConfigurationProvider : ILocalService
+    public interface IConfigurationProvider : Core.Binding.IConfigurationProvider, ILocalService
```

Comment:
I was really debating if I should just remove `ILocalService` and the assertion, but I didn't want to further changes to VsSessionHost and changing the services dictionary to `object`, it didn't feel right.
So I opted for an interface that implements both  `IConfigurationProvider` and `ILocalService`.
WDYT?

Reply:
I think creating an interface that creates both is an ok compromise, but I don't think it should be called `IConfigurationProvider`. It's very confusing.

We're effectively trying to hide the fact that we have two unrelated approaches to constructing objects, but we can't do it cleanly. All of the service-based usages have to include `using IConfigurationProvider = SonarLint.VisualStudio.Integration.NewConnectedMode.IConfigurationProvider;`, and the MEF export of the implementing class is confusing.

It would be less confusing to make it more explicit what we're doing by giving the new interface a different name e.g. `IConfigurationProviderService`.